packages/devextreme/testing/tests/DevExpress.ui.widgets/chatParts/messageBox.tests.js

```diff
@@ -293,6 +297,237 @@ QUnit.module('MessageBox', moduleConfig, () => {
+            }
+        });
+
+        QUnit.test('should be triggered with correct arguments', function(assert) {
+            assert.expect(4);
+
+            this.reinit({
+                onTypingStart: (e) => {
+                    const { component, element, event } = e;
+
+                    assert.strictEqual(component, this.instance, 'component field is correct');
+                    assert.strictEqual(isRenderer(element), !!config().useJQuery, 'element is correct');
+                    assert.strictEqual($(element).is(this.$element), true, 'element field is correct');
+                    assert.strictEqual(event.type, 'input', 'e.event.type is correct');
+                },
+            });
+
+            keyboardMock(this.$input)
+                .focus()
+                .type('n');
+        });
+    });
+
+    QUnit.module('onTypingEnd event', () => {
+        QUnit.test('should be triggered once if a character is entered in the input', function(assert) {
+            const clock = sinon.useFakeTimers({ now: new Date().getTime() });
+            const onTypingEndStub = sinon.stub();
+
+            this.reinit({ onTypingEnd: onTypingEndStub });
+
+            try {
+                const keyboard = keyboardMock(this.$input);
+
+                keyboard
+                    .focus()
+                    .type('n');
+
+                assert.strictEqual(onTypingEndStub.callCount, 0, 'is not called immediately after entering');
+
+                clock.tick(TYPING_END_DELAY - 500);
+
+                assert.strictEqual(onTypingEndStub.callCount, 0, 'is not called still');
+
+                clock.tick(500);
+
+                assert.strictEqual(onTypingEndStub.callCount, 1, 'is called once after delay');
+            } finally {
+                clock.restore();
+            }
+        });
+
+        QUnit.test('should be possible to update it at runtime', function(assert) {
+            const clock = sinon.useFakeTimers({ now: new Date().getTime() });
+            const onTypingEndStub = sinon.stub();
+
+            try {
+                this.instance.option('onTypingEnd', onTypingEndStub);
+
+                keyboardMock(this.$input)
+                    .focus()
+                    .type('n');
+
+                clock.tick(TYPING_END_DELAY);
+
+                assert.strictEqual(onTypingEndStub.callCount, 1);
+            } finally {
+                clock.restore();
+            }
+        });
+
+        QUnit.test('should not be called if the user continues to enter text during the delay', function(assert) {
+            const clock = sinon.useFakeTimers({ now: new Date().getTime() });
+            const onTypingEndStub = sinon.stub();
+
+            this.reinit({ onTypingEnd: onTypingEndStub });
+
+            try {
+                const keyboard = keyboardMock(this.$input);
+
+                keyboard
+                    .focus()
+                    .type('n');
+
+                clock.tick(TYPING_END_DELAY - 1);
```

Comment:
Let's increase timeout sensitivity to 10 & 20 instead of 1 & 2.

Reply:
Updated